tests/assets/move-projects/car-wash-example/sources/CarWash.move

```diff
@@ -0,0 +1,92 @@
+/// This module defines a minimal Coin and Balance.
+module DeveloperBob::CarWash {
+    use std::signer;
+    use substrate::balance;
+
+    /// Address of the owner of this module
+    const MODULE_OWNER: address = @DeveloperBob;
+
+    /// Simple solution, fixed price for one washing coin.
+    const COIN_PRICE: u128 = 1000000000000000000; // equals 1 UNIT
+
+    /// Error codes
+    const NOT_MODULE_OWNER: u64 = 0;
+    const MODULE_NOT_INITIALIZED: u64 = 1;
+    const MODULE_ALREADY_INITIALIZED: u64 = 2;
+    const USER_ALREADY_EXISTS: u64 = 3;
+    const USER_DOES_NOT_EXIST: u64 = 4;
+    const NO_COINS_AVAILABLE: u64 = 5;
+    const COIN_LIMIT_REACHED: u64 = 6;
+
+    /// Struct stores number of coins for each user.
+    struct Balance has key, store {
+        coins: u8
+    }
+
+    /// Method executes the ICO without money. The module owner (also car wash owner) gets deposited the minted washing coins.
+    public fun initial_coin_minting(module_owner: &signer) {
+        // Only the owner of the module can initialize this module
+        assert!(signer::address_of(module_owner) == MODULE_OWNER, NOT_MODULE_OWNER);
+
+        // Do not initialize the module twice!
+        assert!(!exists<Balance>(signer::address_of(module_owner)), MODULE_ALREADY_INITIALIZED);
+
+        // Deposit maximum number of coins to module owner's account. 
+        move_to(module_owner, Balance { coins: 255 });
+    }
+
+    /// Registers a new user. The account address will be added to storage Balance with zero initial washing coins.
+    public fun register_new_user(account: &signer) {
+        // Verify user does not already exist.
+        assert!(!exists<Balance>(signer::address_of(account)), USER_ALREADY_EXISTS);
+
+        // Store new account with zero coins to storage.
+        move_to(account, Balance { coins: 0 });
+    }
+
+    /// Transfers `amount` of tokens from `from` to `to`.
+    public fun buy_coin(user: &signer) acquires Balance {
```

Comment:
I wouldn't implemented it, but this note would be good enough - the example is perfect for our showcase
```suggestion
    // Note: It would be nice to have the ability to buy multiple coins at once, but it's not implemented for this example.
    public fun buy_coin(user: &signer) acquires Balance {
```

Reply:
Done :blush: